common/app/navigation/helpers/UrlHelpers.scala

```diff
@@ -1,9 +1,103 @@
+  case object SlimHeaderDropdown extends Position
+  // SideMenu is the full menu you get from clicking the hamburger icon on mobile
+  case object SideMenu extends Position
+  case object AmpHeader extends Position
+  case object Footer extends Position
+
+  def getCampaignCode(destination: ReaderRevenueSite, position: Position)(implicit request: RequestHeader): Option[String] = {
+    val isInHeaderTestControlGroup = mvt.ABNewDesktopHeaderControl.isParticipating
+    val editionId = Edition(request).id
+
+    (destination, position, isInHeaderTestControlGroup) match {
+      case (Membership, NewHeader | SideMenu, _) => Some(s"mem_${editionId.toLowerCase}_web_newheader")
+      case (Membership, OldHeader, true) => Some(s"mem_${editionId.toLowerCase}_web_newheader_control")
+      case (Membership, OldHeader, false) => Some(s"DOTCOM_HEADER_BECOMEMEMBER_${editionId.toUpperCase}")
+      case (Membership, AmpHeader, _) => Some("AMP_HEADER_GU_SUPPORTER")
+      case (Membership, SlimHeaderDropdown, _) => Some(s"NGW_TOPNAV_${editionId.toUpperCase}_GU_MEMBERSHIP")
+
+      // TODO: double-check new & old header are the same for this one?
+      case (Contribute, NewHeader | OldHeader, _) => Some("gdnwb_copts_co_dotcom_header")
+      case (Contribute, Footer, _) => Some("gdnwb_copts_memco_dotcom_footer")
+
+      case (Subscribe, SideMenu, _) => Some(s"NGW_NEWHEADER_${editionId}_GU_SUBSCRIBE")
+      case (Subscribe, NewHeader, _) => Some(s"subs_${editionId}_web_newheader")
+      case (Subscribe, OldHeader, true) => Some(s"subs_${editionId}_web_newheader_control")
+      case (Subscribe, OldHeader, false) => Some(s"NGW_HEADER_${editionId}_GU_SUBSCRIBE")
+      case (Subscribe, SlimHeaderDropdown, _) => Some(s"NGW_TOPNAV_${editionId.toUpperCase}_GU_SUBSCRIBE")
+
+      case (_, _, _) => None
+    }
+  }
+
+  case class ABTest(name: String, variant: String)
+
+  def getHeaderABTestInfo(implicit request: RequestHeader): Option[ABTest] =
+    if (mvt.ABNewDesktopHeaderControl.isParticipating) {
+      Some(ABTest("NewDesktopHeader", "control"))
+    } else if (mvt.ABNewDesktopHeaderVariant.isParticipating) {
+      Some(ABTest("NewDesktopHeader", "variant"))
+    } else {
+      None
+    }
+
+  def getReaderRevenueUrl(destination: ReaderRevenueSite, position: Position)(implicit request: RequestHeader): String = {
+    val campaignCode = getCampaignCode(destination, position)
+    val abTest = position match {
+      case NewHeader | OldHeader => getHeaderABTestInfo
+      case _ => None
+    }
+
+    val acquisitionData = Json.obj(
+      "source" -> AcquisitionSource.GuardianWeb.name,
+      "componentId" -> campaignCode,
+      "componentType" -> (position match {
+        case NewHeader | OldHeader | AmpHeader | SideMenu | SlimHeaderDropdown => ComponentType.AcquisitionsHeader
+        case Footer => ComponentType.AcquisitionsFooter
+      }).name,
+      // TODO: there's no way to get this serverside is there? replace clientside??
```

Comment:
You can't get this information reliably server side since most of the requests are cached